docs/gc.md

```diff
@@ -93,23 +117,33 @@ More information about Concurrent Scavenge mode can be found in the blog post [C
 
+#### GC operations
+
+
 During VM startup, the GC divides the heap memory into regions of equal size. These regions remain static for the lifetime of the VM and are the basic unit of garbage collection and allocation operations. For example, when the heap is expanded or contracted, the memory committed or released corresponds to a certain number of regions. Although the Java heap is a contiguous range of memory addresses, any region within that range can be committed or released as required. This enables the Balanced GC to contract the heap more dynamically and aggressively than other garbage collectors, which typically require the committed portion of the heap to be contiguous.
 
 Regions impose a maximum object size. Objects are always allocated within the bounds of a single region and are never permitted to span regions. The region size is always a power of two; for example, 512 KB, 1 MB, and so on (where KB is 2<sup>10</sup> bytes and MB is 2<sup>20</sup> bytes). The region size is selected at startup based on the maximum heap size. The collector chooses the smallest power of two which will result in less than 2048 regions, with a minimum region size of 512 KB. Except for small heaps (less than about 512 MB) the VM aims to have between 1024 and 2047 regions.
 
-For further information see [Balanced Garbage Collection policy](https://www.ibm.com/support/knowledgecenter/SSYKE2_8.0.0/com.ibm.java.vm.80.doc/docs/mm_gc_balanced.html).
+Object ages are tracked for each region with a maximum of 24 possible generations. The following diagram illustrates the structure of the object heap:
+
+![The diagram is explained in the surrounding text](./cr/balanced_regions.png "Region structure and characteristics found in the object heap")
+
+ An age 0 region, known as the *eden* space, contains the newest objects allocated. Regions of intermediate age serve as survivor regions to receive live objects copied forwarded from younger regions. When the eden space is full, a partial garbage collection (PGC) runs to reclaim space, typically by using a *copy forward* operation. In specific cases, a *mark-compact* operation might be used, for example, when there are not enough free survivor regions available. The PGC is a stop-the-world operation, after which new regions are allocated to create a new eden space. A PGC always includes the eden space, but might include older regions. Objects within collectible regions of intermediate age N are evacuated to survivor regions of age N+1. Regions of age 24 are never included in PGC collection sets but might be included for compaction in global GC (GGC) cycles, which mark and sweep the whole heap and compact regions that need defragmentation.
```

Comment:
We typically keep copy-forward together leading to past tense copy-forwarded. But for simplicity you can just say copied or even moved.

Reply:
Actually, regions of age 24 are also be included in PGC (in fact in on farily regular basis), for the same reason: to de-fragment them. That's the biggest difference between Gencon and Balanced (since Scavenge cannot ever collect any part of Tenure). We want to incrementally de-fragment old regions instead of waiting for them to become so fragmented that we are forced to do a global compact (that can be very long).
They of course stay at the age 24.

Global GC in Balanced is very rare. It will occur in very tight memory conditions, when PGC and concurrent GMP cannot keep up, or as a desperate attempt to recover last bits of free memory prior throwing Out Of Memory exceptions.
Global GC are also performed as explict GCs, such as those invoked from application thorugh System.gc().

Reply:
Eden is a _set_ of regions of age 0.

Reply:
Updates complete. Staging underway.